Tweak.xm

```diff
@@ -1,173 +1,129 @@
-CGPoint dockPoint;
+#include <UIKit/UIKit.h>
+#include "SpringBoardHome.h"
+
+BOOL isEnabled;
 BOOL shouldHide;
+BOOL initialized;
 
-@interface SBDockView : UIView
-@end
-
-@interface SBRootFolderView : UIView
-@property (nonatomic, assign, readwrite) NSUInteger dockEdge;
-@property (nonatomic,readonly) SBDockView *dockView; 
-@end
-
-%group ios13
-
-%hook SBRootFolderView
-
--(void)updateIconListIndexAndVisibility:(BOOL)arg1{
-    if (@available(ios 13, *)) {
-        self.dockEdge = 1;
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0) {
-            self.dockView.frame = CGRectMake(0, 50, self.dockView.frame.size.width, self.dockView.frame.size.height);
-            if (!dockPoint.x) {
-                dockPoint = CGPointMake(self.dockView.center.x, (self.dockView.center.y + 50));
-            }
-            self.dockView.center = dockPoint;
-        }
-        %orig;
-    } else {
-		%orig;
-	}
-}
+#define safeAreaInsets [[[UIApplication sharedApplication] keyWindow] safeAreaInsets]
 
+#define kIconController [%c(SBIconController) sharedInstance]
+#define kRootFolderController [kIconController _rootFolderController]
+#define kIconModel [kIconController model]
+#define kDockView kRootFolderController.rootFolderView.dockView
 
-%end
+#define boolValueForKey(prefs, key, defaultValue) ((prefs && [prefs objectForKey:key]) ? [[prefs objectForKey:key] boolValue] : defaultValue)
+
+%hook SBRootFolderController
 
-%hook SBDockView
+-(void)viewDidLoad
+{
+    %orig;
 
--(CGRect)frame{
-    if (shouldHide) {
-        [self setHidden:TRUE];
+    if (!isEnabled)
+    {
+        self.dockEdge = 4;
+        return;
     }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            CGRect origFrame = %orig;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            return fixedFrame;
-        } else {
-            return %orig;
-        }
-    } else {
-		return %orig;
-	}
+
+    self.dockEdge = 1;
+    self.rootFolderView.dockView.hidden = shouldHide;
+    [[self.rootFolderView dockView].superview bringSubviewToFront:[self.rootFolderView dockView]];
```

Comment:
I kick this to the front here because using layoutInsets to properly move the root folder icons causes the SBIconScrollView to block touches to SBDockView.

Moving SBDockView to the top fixes this rather cleanly.